pkg/scheduler/dependency_analyzer.go

```diff
@@ -133,8 +147,23 @@ func assignVertex(vertex, root string, assigned map[string]bool, components map[
 	}
 }
 
-func EnsureNoCycles(dependencies []client.Dependency) error {
-	cycles := detectCycles(dependencies)
+func EnsureNoCycles(dependencies []client.Dependency, resdefs map[string]client.ResourceDefinition) error {
+	flows := map[string]*client.Flow{}
+	if _, found := resdefs["flow/"+interfaces.DefaultFlowName]; !found {
+		flows["flow/"+interfaces.DefaultFlowName] = newDefaultFlowObject()
+	}
+	for _, v := range resdefs {
+		if v.Flow != nil {
+			flows["flow/"+v.Flow.Name] = v.Flow
+		}
+	}
+
+	var cycles [][]string
+	for _, flow := range flows {
+		for useDestructionSelection := false; !useDestructionSelection; useDestructionSelection = !useDestructionSelection {
```

Comment:
I don't see any sense in this for. I suggest removing it completely (just leave `cycles = append(cycles, detectCycles(dependencies, flows, flow, false)...)` below.